plugins/modules/source_control/gitlab/gitlab_user.py

```diff
@@ -412,13 +467,57 @@ def createUser(self, arguments):
         if self._module.check_mode:
             return True
 
+        identities = None
+        if 'identities' in arguments:
+            identities = arguments['identities']
+            del arguments['identities']
+
         try:
             user = self._gitlab.users.create(arguments)
+            if identities:
+                self.addIdentities(user, identities)
+
         except (gitlab.exceptions.GitlabCreateError) as e:
             self._module.fail_json(msg="Failed to create user: %s " % to_native(e))
 
         return user
 
+    '''
+    @param user User object
+    @param identites List of identities to be added/updated
+    @param overwrite_identities Overwrite user identities with identities passed to this module
+    '''
+    def addIdentities(self, user, identities, overwrite_identities=False):
+        if self._module.check_mode:
+            return True
```

Comment:
You need to return `True` only when non-checkmode would also do. It's best to simply skip the `user.save()` step when `self._module.check_mode` is truish.